waku/node/waku_node.nim

```diff
@@ -223,31 +226,101 @@ proc registerRelayDefaultHandler(node: WakuNode, topic: PubsubTopic) =
     await filterHandler(topic, msg)
     await archiveHandler(topic, msg)
 
-  node.wakuRelay.subscribe(topic, defaultHandler)
-
+  discard node.wakuRelay.subscribe(topic, defaultHandler)
 
 proc subscribe*(node: WakuNode, topic: PubsubTopic) =
+  ## Subscribe to a pubsub topic without a handler.
+  ## Adds default handler to push messages to FITLER & STORE
+  
   if node.wakuRelay.isNil():
     error "Invalid API call to `subscribe`. WakuRelay not mounted."
     return
 
+  if node.handlersTable.hasKey(topic):
+    error "Invalid API call to `subscribe`. Was already subscribed"
+    return
+
   debug "subscribe", pubsubTopic= topic
 
   node.topicSubscriptionQueue.emit(SubscriptionEvent(kind: PubsubSub, pubsubSub: topic))
+
+  if node.subscriptions.hasKeyOrPut(topic, (true, 0)):
+    node.subscriptions.mgetOrPut(topic, (true, 0)).pubsubTopic = true
+
   node.registerRelayDefaultHandler(topic)
 
+proc autoSubscribe*(node: WakuNode, topic: ContentTopic) =
+  ## Subscribe to a content topic without a handler.
+  ## Adds default handler to push messages to FITLER & STORE
+  
+  if node.wakuRelay.isNil():
+    error "Invalid API call to `autoSubscribe`. WakuRelay not mounted."
+    return
+
+  if node.handlersTable.hasKey(topic):
+    error "Invalid API call to `autoSubscribe`. Was already subscribed"
+    return
+
+  debug "autoSubscribe", contentTopic= topic
+
+  let shard = getShard(topic).valueOr:
+    error "Autosharding error: ", error=error
+    return
+
+  node.topicSubscriptionQueue.emit(SubscriptionEvent(kind: PubsubSub, pubsubSub: shard))
```

Comment:
@SionoiS, @jm-clius : After our discussion today, here is a good point to mention or rather pop up with.
Why we just simply detach subscribe/unsubscribe from the protocol. Due we have now different sub/unsub functions in waku_node, at least separate for relay and for filter (+ sharding for both).
I think we do know the use case from just the arguments. Is this true?
If so we could hide all these complexities behind one sub/unsub interface.

In the sense we can assume the user of the node (operator of the node) does know what type of service he/she would like to run.
Hm?

Same way I'm thinking of the discussed topic of message handler callbacks. Does it really necessary for the application level to define exactly which at the level of protocol that I'm interested in this or that protocol's messages? Is it a real use case from application point of view to be interested in relayed messages parallel with using filtered (lightweight) messages?

Maybe, if I miss something from a bigger picture, forgive me, plz! :-)

Reply:
@NagyZoltanPeter We should discuss this next meeting. would be a big change but maybe worth it?

Reply:
@NagyZoltanPeter good point. Indeed where I think we'd like to get to. What makes this a bit more complicated is simply that applications expect a bit more from the `filterSubscribe` - they would like to specify exactly which service node to subscribe to and, importantly, would have to subscribe to multiple service nodes for service reliability. This implies that the application should also deduplicate messages for filter subscriptions etc. However, I agree that all this functionality can eventually be abstracted away from the application where we simply provide a subscribe/unsubscribe on either content topic or pubsub topic with a single handler for received messages, no matter the origin. Perhaps we'd then use some other means to determine if a content topic subscription should be a "light" subscribe (filter) or a "full" subscribe (relay). However, since we still have these complexities I'd still separate the API for "full" subscribe and "light" subscribe until we have a more infrastructure to help applications here.

Reply:
Yes, I know it could be done only with a middle to bigger architectural change.